### PR TITLE
pytests for pymcfost

### PR DESCRIPTION
**Pull request by purmortal**
This PR is for building workflow for testing functions in pymcfost using pytest.
Changes include:
1. Create `test_build.yml` to run the test workflow, python versions from 3.8 to 3.11 in both Linux and MacOS will be tested.
2. Create `test_plot.py` to test if the plotted figures are the same as the expected ones, codes are based on the jupyter notebook tutorial provided in this repo.
3. Expected figures with corresponding data (> 70 MB in total) used to load into pymcfost are zipped into a `tar.gz` file on a remote server, which is to be maintained for future changes. Command to download the data has been integrated into  `test_plot.py` when running the workflow.
4. Previously existing test*.py scripts are also integrated into the workflow.

### CODE REVIEW

File: .github/workflows/test_build.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Install pymcfost with dependecies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install .
+
+      - name: Install pytest
+        run: pip install pytest
+
+      - name: Run tests
```
Comment (conradtchan) [inline .github/workflows/test_build.yml:37]: pytest will automatically look in a directory named "tests" and run every file matching test_*.py. So you can just replace those 3 `pytest tests/...` lines with
```
pytest
```

**Review by conradtchan — CHANGES_REQUESTED** [review]

File: .github/workflows/test_build.yml (hunk 1 above)
Comment (purmortal) [reply .github/workflows/test_build.yml:37]: Done

**Comment by conradtchan** [comment]
Thanks @purmortal, looks good! Once @cpinte has hosted the reference data, we can update it in the workflow and merge this in.

**Comment by purmortal** [comment]
> Thanks @purmortal, looks good! Once @cpinte has hosted the reference data, we can update it in the workflow and merge this in.

Great, thanks so much for your help!

**Comment by cpinte** [comment]
I have uploaded the data here : http://ipag.osug.fr/public/pintec/pymcfost/testdata.tar.gz

**Comment by purmortal** [comment]
> I have uploaded the data here : http://ipag.osug.fr/public/pintec/pymcfost/testdata.tar.gz

Hi Chris, I have updated the files using the new link. All the tests passed.

**Review by cpinte — APPROVED** [review]
Thanks!

**Pull request merged**